src/NuGet.Core/NuGet.Build.Tasks.Console/Program.cs

```diff
@@ -174,5 +216,74 @@ private static bool TryParseArguments(string[] args, out (Dictionary<string, str
                 return false;
             }
         }
+
+        /// <summary>
+        /// Logs an error to the specified <see cref="TextWriter" />.
+        /// </summary>
+        /// <param name="errorWriter">The <see cref="TextWriter" /> to write the error to.</param>
+        /// <param name="format">The formatted string of the error.</param>
+        /// <param name="args">An object array of zero or more objects to format with the error message.</param>
+        /// <returns><see langword="false" /></returns>
+        private static void LogError(TextWriter errorWriter, string format, params object[] args)
+        {
+            errorWriter.WriteLine(format, args);
+        }
+
+        /// <summary>
+        /// Attempts to deserialize global properties from the standard input stream.
+        /// </summary>
+        /// <remarks>
+        /// <param name="errorWriter">A <see cref="TextWriter" /> to write errors to if one occurs.</param>
+        /// <param name="reader">The <see cref="BinaryReader" /> to use when deserializing the arguments.</param>
+        /// <param name="globalProperties">Receives a <see cref="Dictionary{TKey, TValue}" /> representing the global properties.</param>
+        /// <returns><see langword="true" /> if the arguments were successfully deserialized, otherwise <see langword="false" />.</returns>
+        internal static bool TryDeserializeGlobalProperties(TextWriter errorWriter, BinaryReader reader, out Dictionary<string, string> globalProperties)
+        {
+            globalProperties = null;
+
+            int count = 0;
+
+            try
+            {
+                // Read the first integer from the stream which is the number of global properties
+                count = reader.ReadInt32();
+            }
+            catch (Exception e)
+            {
+                LogError(errorWriter, Strings.Error_StaticGraphRestoreArgumentsParsingFailedExceptionReadingStream, e.Message, e.ToString());
+
+                return false;
+            }
+
+            // If the integer is negative or greater than or equal to int.MaxValue, then the integer is invalid.  This should never happen unless the bytes in the stream contain completely unexpected values
+            if (count < 0 || count >= int.MaxValue)
```

Comment:
`BinaryReader.ReadInt32();` docs state that this method returns a 4-byte signed integer. Given that `count` is of type integer, the chance of it's value being greater than `int.MaxValue` is 0, how about changing the line of code to `if (count < 0)`.

Reply:
I had to bring this back.  If the count is too large, then the code will attempt to new up a `Dictionary<TKey, TValue>` with that many items and the constructor can throw an `OutOfMemoryException`.  I'm placing an upper bound of `short.MaxValue` to guard against too many global properties.  There is no way any user will ever have 32,000+ global properties to a value that large in the byte stream is considered a problem.